packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts

```diff
@@ -226,15 +227,32 @@ class DefaultMutationMapperStore implements MutationMapperStore {
         }
     }
 
-    public getAnnotatedMutations(transcriptId: string): Mutation[] {
+    @memoize
+    protected getAnnotatedMutationsByTranscriptId(
+        mutations: Mutation[],
+        transcriptId: string,
+        indexedVariantAnnotations: {
+            [genomicLocation: string]: VariantAnnotation;
+        }
+    ) {
         // by default annotate every mutation, no check for canonical transcript
-        return this.indexedVariantAnnotations.result
-            ? getMutationsByTranscriptId(
-                  this.getMutations(),
-                  transcriptId,
-                  this.indexedVariantAnnotations.result
-              )
-            : this.getMutations();
+        return getMutationsByTranscriptId(
+            mutations,
+            transcriptId,
+            indexedVariantAnnotations
+        );
+    }
+
+    public getAnnotatedMutations(transcriptId: string): Mutation[] {
```

Comment:
lets take this opportunity to comment this code.  like it doesn't make a lot of sense to me.  if indexedVariantAnnotations is not complete yet, then we're just going to return un-annotated mutations?  that doesn't feel right.   i would think we would not want to use mutations before we know whether or not they are annotated.   that's an argument for passing indexedVariantAnnotations.result in as an argument.

Reply:
Right, this might confusing. I can't really remember why we did it this way, but there might be cases where we don't have any variant annotation data. I'll try to figure out the reason and add a comment for this code.